java/solrmarc_mixin/src/de/unituebingen/ub/ubtools/solrmarcMixin/TuelibMixin.java

```diff
@@ -1120,6 +1120,19 @@ private boolean isInFactPrimaryAuthor(final List<Subfield> subfieldFields4) {
     }
 
 
+    public Set<String> getAuthorPPNs(final Record record, final String fieldSpecs) {
+        final Set<String> values = SolrIndexer.instance().getFieldList(record, fieldSpecs);
+        final Set<String> ppns = new TreeSet<>();
+
+        for (final String value : values) {
+            if (value.startsWith(ISIL_PREFIX_BSZ))
+                ppns.add(value.substring(0, ISIL_PREFIX_BSZ.length()));
```

Comment:
I suspect that you should use value.substring(ISIL_PREFIX_BSZ.length()) instead.